ASoC: SOF: Change SND_SOC_SOF_TOPLEVEL from config to menuconfig

ujfalusi opened the pull request:
We have growing number of options under SND_SOC_SOF_TOPLEVEL and it makes
the selection of the options much obvious and cleaner if the SOF config
is presented under a separate page.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

ujfalusi commented:
To show the difference, please consider the following screenshots (x86_64 config on my machine):
Current config screen for ASoC, including SOF:
![image](https://user-images.githubusercontent.com/12640614/132818488-05e8ff39-1f05-4e1d-9a38-626777471265.png)

With the change to menuconfig the ASoC page (SOF is highlighted):
![image](https://user-images.githubusercontent.com/12640614/132818626-69a2d2ac-dd2a-4497-b254-b4c33bc25072.png)

and when entering to SOF menu:
![image](https://user-images.githubusercontent.com/12640614/132818658-330e00e8-5301-48fc-b725-813eed1951ae.png)

plbossart reviewed (approved):
@ujfalusi You may also want to add a mention of 4 companies contributing to SOF (Intel, NXP, AMD, Mediatek) to further justify the submenu.

ujfalusi commented:
> @ujfalusi You may also want to add a mention of 4 companies contributing to SOF (Intel, NXP, AMD, Mediatek) to further justify the submenu.

@plbossart, true. I will update the PR on Monday, thanks for the suggestion.

ujfalusi commented:
Changes since v1:
- change the commit message (and the subject) to mention the growing number of companies contributing, Thanks for the suggestion, @plbossart

bardliao reviewed (approved):
Thanks @ujfalusi I like the change.
